### PR TITLE
feat(unity): IL2CPP

### PR DESCRIPTION
**Pull request by bitsandfoxes**
IL2CPP feature docs.

### CODE REVIEW

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+title: IL2CPP Line Numbers
+description: "Learn how the SDK provides line numbers to issues from IL2CPP builds."
+sidebar_order: 10
+---
+
+<Note>
+
+The IL2CPP line number support relies on the debug symbol upload to function properly. With the automated debug symbol upload enabled, the Sentry SDK will upload the line mapping through Sentry CLI.
+
+The feature is available from Unity versions `2020.3` and `2021.3` onwards. We running tests in CI against the current LTS, starting from `2020.3.37f1` and `2021.3.6f1`.
```
Comment (marandaneto) [inline src/platforms/unity/configuration/il2cpp.mdx:11]:
```suggestion
The feature is available from Unity versions `2020.3` and `2021.3` onwards. We are running tests in CI against the current LTS, starting from `2020.3.37f1` and `2021.3.6f1`.
```

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Unity provides you the option to select [IL2CPP](https://docs.unity3d.com/Manual/IL2CPP.html) as scripting backend. This leads to C++ code being generated from your compiled C# (IL) code, hence the name `IL to CPP`.
+
+The Sentry SDK provides line number support for IL2CPP builds for the following platforms:
+
+- Desktop: Windows, macOS, Linux
+- Android
+- iOS
+
+When an error happens in your game, from its perspective, that errors happens in native code.
+The mapping between your C# code and the generated C++ code is part of the build process and does not get shipped with with your game. This leads to the event the Sentry SDK reports can only containing the native stacktrace. To be able to provide you with C# line numbers, the Sentry server needs to have access to that line mapping.
```
Comment (marandaneto) [inline src/platforms/unity/configuration/il2cpp.mdx:24]:
```suggestion
When an error happens in your game, from its perspective, that error happens in native code.
The mapping between your C# code and the generated C++ code is part of the build process and does not get shipped with your game. This leads to the event the Sentry SDK reports can only contain the native stacktrace. To be able to provide you with C# line numbers, the Sentry server needs to have access to that line mapping.
```

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+The feature is currently experimental and opt-in through the editor configuration window `Advanced -> IL2CPP line numbers` or programmatically
+
+```csharp
+
+options.Il2CppLineNumberSupportEnabled = true;
+
+```
```
Comment (marandaneto) [inline src/platforms/unity/configuration/il2cpp.mdx:32]:
```suggestion
```csharp
options.Il2CppLineNumberSupportEnabled = true;
```
```

Comment (marandaneto) [inline src/platforms/unity/configuration/il2cpp.mdx:32]: You can remove the extra lines, the suggestion does not work with back tips.

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+The Sentry SDK provides line number support for IL2CPP builds for the following platforms:
+
+- Desktop: Windows, macOS, Linux
+- Android
+- iOS
+
+When an error happens in your game, from its perspective, that errors happens in native code.
+The mapping between your C# code and the generated C++ code is part of the build process and does not get shipped with with your game. This leads to the event the Sentry SDK reports can only containing the native stacktrace. To be able to provide you with C# line numbers, the Sentry server needs to have access to that line mapping.
+
+The feature is currently experimental and opt-in through the editor configuration window `Advanced -> IL2CPP line numbers` or programmatically
```
Comment (marandaneto) [inline src/platforms/unity/configuration/il2cpp.mdx:26]: I'd add the sentence that we always do:

> Feature is available in Beta. Beta features are still in-progress and may have bugs. We recognize the irony.

File: src/platforms/unity/configuration/il2cpp.mdx (hunk 3 above)
Comment (bitsandfoxes) [reply src/platforms/unity/configuration/il2cpp.mdx]: Thanks for the hint!

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+Unity provides you the option to select [IL2CPP](https://docs.unity3d.com/Manual/IL2CPP.html) as scripting backend. This leads to C++ code being generated from your compiled C# (IL) code, hence the name `IL to CPP`.
+
+The Sentry SDK provides line number support for IL2CPP builds for the following platforms:
+
+- Desktop: Windows, macOS, Linux
+- Android
+- iOS
+
+When an error happens in your game, from its perspective, that error happens in native code.
+The mapping between your C# code and the generated C++ code is part of the build process and does not get shipped with your game. This leads to the event the Sentry SDK reports can only contain the native stacktrace. To be able to provide you with C# line numbers, the Sentry server needs to have access to that line mapping.
```
Comment (bruno-garcia) [inline src/platforms/unity/configuration/il2cpp.mdx:30]:
```suggestion
The mapping between your C# code and the generated C++ code is part of the build process and does not get shipped with your game. This leads to the event the Sentry SDK reports can only contain the native stacktrace. To be able to provide you with C# line numbers, the Sentry server needs to have access to that line mapping through [Debug Information Files](https://docs.sentry.io/platforms/unity/data-management/debug-files/).
```

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+You can opt-into this feature through the editor configuration window `Advanced -> IL2CPP line numbers` or programmatically
```
Comment (bruno-garcia) [inline src/platforms/unity/configuration/il2cpp.mdx:32]:
```suggestion
You can opt-into this feature through the editor configuration window `Tools -> Sentry -> Advanced -> IL2CPP line numbers` or [programmatically](/platforms/unity/configuration/options/#programmatic-configuration).
```

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+---
+title: IL2CPP Line Numbers
+description: "Learn how the SDK provides line numbers to issues from IL2CPP builds."
+sidebar_order: 10
+---
+
+<Note>
+
+The IL2CPP line number support is available in Beta. Beta features are still in-progress and may have bugs. We recognize the irony.
```
Comment (imatwawana) [inline src/platforms/unity/configuration/il2cpp.mdx:9]:
```suggestion
IL2CPP line number support is available in Beta. Beta features are still in-progress and may have bugs. We recognize the irony.
```

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+</Note>
+
+<Note>
+
+The IL2CPP line number support relies on the debug symbol upload to function properly. With the automated debug symbol upload enabled, the Sentry SDK will upload the line mapping through Sentry CLI.
```
Comment (imatwawana) [inline src/platforms/unity/configuration/il2cpp.mdx:15]:
```suggestion
IL2CPP line number support relies on the debug symbol upload to function properly. With the automated debug symbol upload enabled, the Sentry SDK will upload the line mapping through Sentry CLI.
```

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+The feature is available from Unity versions `2020.3` and `2021.3` onwards. We are running tests in CI against the current LTS, starting from `2020.3.37f1` and `2021.3.6f1`.
+
+</Note>
+
+Unity provides you the option to select [IL2CPP](https://docs.unity3d.com/Manual/IL2CPP.html) as scripting backend. This leads to C++ code being generated from your compiled C# (IL) code, hence the name `IL to CPP`.
```
Comment (imatwawana) [inline src/platforms/unity/configuration/il2cpp.mdx:21]:
```suggestion
Unity provides you the option to select [IL2CPP](https://docs.unity3d.com/Manual/IL2CPP.html) as your scripting backend. This leads to C++ code being generated from your compiled C# (IL) code, hence the name `IL to CPP`.
```

File: src/platforms/unity/configuration/il2cpp.mdx (hunk 5 above)
Comment (imatwawana) [inline src/platforms/unity/configuration/il2cpp.mdx:30]:
```suggestion
The mapping between your C# code and the generated C++ code is part of the build process and isn't shipped with your game. That means the event the Sentry SDK reports can only contain the native stack trace. To provide you with C# line numbers, the Sentry server needs to have access to that line mapping.
```

File: src/platforms/unity/configuration/il2cpp.mdx (hunk 6 above)
Comment (imatwawana) [inline src/platforms/unity/configuration/il2cpp.mdx:32]:
```suggestion
You can opt in to this feature through the editor configuration window by selecting `Advanced -> IL2CPP line numbers` or programmatically:
```

Comment (bruno-garcia) [reply src/platforms/unity/configuration/il2cpp.mdx:32]: The anchor depends on PR: https://github.com/getsentry/sentry-docs/pull/5338

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+The Sentry SDK provides line number support for IL2CPP builds for the following platforms:
+
+- Desktop: Windows, macOS, Linux
+- Android
+- iOS
+
+When an error happens in your game, from its perspective, that error happens in native code.
```
Comment (imatwawana) [inline src/platforms/unity/configuration/il2cpp.mdx:29]: It's odd to talk about the game's perspective like it's a person. Can we rephrase this?

```suggestion
When an error happens in your game, from its perspective, that error happens in native code.
```

**Review by bruno-garcia — APPROVED** [review]
A couple of suggestions otherwise LGTM!!

**Review by imatwawana — COMMENTED** [review]
Made some wording edits. I also asked about the phrasing of one of the sentences.

File: src/platforms/unity/configuration/il2cpp.mdx (hunk 10 above)
Comment (bitsandfoxes) [reply src/platforms/unity/configuration/il2cpp.mdx]: Sure! I rephrased it.

**Review by imatwawana — APPROVED** [review]

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+The feature is available from Unity versions `2020.3` and `2021.3` onwards. We are running tests in CI against the current LTS, starting from `2020.3.37f1` and `2021.3.6f1`.
+
+</Note>
+
+Unity provides you the option to select [IL2CPP](https://docs.unity3d.com/Manual/IL2CPP.html) as your scripting backend. This leads to C++ code being generated from your compiled C# (IL) code, hence the name `IL to CPP`.
+
+The Sentry SDK provides line number support for IL2CPP builds for the following platforms:
+
+- Windows (*currently limited to Unity 2021.3)
```
Comment (bruno-garcia) [inline src/platforms/unity/configuration/il2cpp.mdx:25]:
```suggestion
- Windows (*currently limited to Unity 2021.3 or newer)
```

**Review by bruno-garcia — APPROVED** [review]

File: src/platforms/unity/configuration/il2cpp.mdx [hunk 12]
```diff
@@ -0,0 +1,37 @@
+---
+title: IL2CPP Line Numbers
+description: "Learn how the SDK provides line numbers to issues from IL2CPP builds."
+sidebar_order: 10
+---
+
+<Note>
+
+IL2CPP line number support is available in Beta. Beta features are still in-progress and may have bugs. We recognize the irony.
```
Comment (bruno-garcia) [inline src/platforms/unity/configuration/il2cpp.mdx:9]:
```suggestion
IL2CPP line number support is available in Beta and requires version 0.22.0 or later of the Sentry SDK for Unity. Beta features are still in-progress and may have bugs. We recognize the irony.
```

**Pull request merged**